### PR TITLE
Make bl/gc and jeebies use the common error dialog

### LINKED ISSUE #157: Consolidate the several tool error windows
**Describe the solution you'd like**
There are several different error windows, e.g. bookloupe/gutcheck, jeebies, etc. which have their own subroutines, dialogs and slightly different behaviour (? not necessarily designed as such). Each one also has its own geometry stored
Combine into using one window with common code where possible

**Describe alternatives you've considered**
Leaving well alone :)

### PR DESCRIPTION
**Pull request by windymilla**
Bookloupe/gutcheck and jeebies each had their own error dialog, with duplicated
code, etc. Now all use the same dialog.
Highlighting code from bl/gc now highlights errors from other tools.

Tests added to test suite for bl/gc and jeebies.

Fixes #157

A separate issue #178 relates to making all line formats the same to give further
consistency and use of common code.

### CODE REVIEW

File: src/lib/Guiguts/ErrorCheck.pm [hunk 1]
```diff
@@ -153,153 +198,204 @@ sub errorcheckpop_up {
...
+            if (
+                    ( not $::verboseerrorchecks )
+                and ( $thiserrorchecktype eq 'W3C Validate CSS' )
+                and (  ( $line =~ /^To show your readers/i )
+                    or ( $line =~ /^Valid CSS Information/i ) )
+            ) {
+                last;
+            }
+            if (
+                ( $line    =~ /^\s*$/i )                                    # skip some unnecessary lines from W3C Validate CSS
```
Comment (cpeel) [inline src/lib/Guiguts/ErrorCheck.pm:231]: While it was like this before, would this comment perhaps be better before the `if` block? As-is it seems to imply it only applies to the current line which is incorrect.

File: src/lib/Guiguts/ErrorCheck.pm [hunk 2]
```diff
@@ -1308,45 +1107,26 @@ sub gutcheck {
...
-    if ( $::gcopt[5] ) { $gutcheckoptions .= 'l' }
-    ;                                    # Line end checking - defaults on
-    if ( $::gcopt[6] ) { $gutcheckoptions .= 'v' }
-    ;                                    # Verbose - list EVERYTHING!
-    if ( $::gcopt[7] ) { $gutcheckoptions .= 'u' }
-    ;                                    # Use file of User-defined Typos
-    if ( $::gcopt[8] ) { $gutcheckoptions .= 'd' }
-    ;                                    # Ignore DP style page separators
+    my $gutcheckoptions = '-ey';    # e - echo queried line. y - puts errors to stdout instead of stderr.
+    $gutcheckoptions .= 't' if $::gcopt[0];    # Check common typos
```
Comment (cpeel) [inline src/lib/Guiguts/ErrorCheck.pm:1115]: This block is *much* more readable now - nicely done.

**Review by cpeel — COMMENTED** [review]
One minor comment, but this looks great.

**Comment by windymilla** [comment]
Thanks - slightly tidied - will do more tidying of same area in other MRs

**Pull request merged**